packages/plans-grid-next/src/lib/test/filter-plan-features-object.ts

```diff
@@ -9,18 +9,21 @@ const feature3 = { getSlug: () => 'feature3' } as FeatureObject;
 const feature4 = { getSlug: () => 'feature4' } as FeatureObject;
 const feature5 = { getSlug: () => 'feature5-jetpack' } as FeatureObject;
 const feature6 = { getSlug: () => 'feature6-jetpack' } as FeatureObject;
+const feature7 = { getSlug: () => 'feature7-conditional' } as FeatureObject;
 
 const planA = {
 	features: {
 		wpcomFeatures: [ feature1, feature2 ],
 		jetpackFeatures: [ feature5, feature6 ],
+		conditionalFeatures: [ feature7 ],
```

Comment:
I am thinking we could explore cleaning up this structure a little. It might even be possible to remove `conditionalFeatures` from `PlanFeaturesForGridPlan` type. It is somewhat ambiguous. These are basically features only added for comparison grid (I think), but those are transformed/put together by respective hook anyhow (`use-restructured-plan-features-for-comparison-grid.ts`). So we are getting different sets for each grid from these hooks already (then maybe no need to re-classify in a separate array this way?).

I haven't looked at these parts in a lot of detail. Just from the surface, I think it might already fail once "conditional Jetpack features" are needed (and if that means adding another structure `conditionalJetpackFeatures`, then it might be time to revisit).

Just thinking out loud, not suggesting a follow up.

Reply:
> I am thinking we could explore cleaning up this structure a little

Yes this stood out to me while making this update. Ideally I don't think a dev should need to understand the categorisation of features (wpcom, jetpack, conditional...) in order to iterate over a plan's features like we are doing here:

https://github.com/Automattic/wp-calypso/blob/b7c5adf0cf88c5c935f983c135c0a894e1ae2a4c/packages/plans-grid-next/src/lib/filter-unused-features-object.ts#L18-L23

There's also the maintenance burden of trying to sync multiple code locations that operate on multiple feature categories if we were to add a new 'category' of features here:

https://github.com/Automattic/wp-calypso/blob/3484df44795298177903d143570b0918a2bae8b3/packages/plans-grid-next/src/types.ts#L35-L40

Perhaps we could get by with a single list of features per plan, and when a particular 'category' of features is needed it can be derived?

I also noticed `features.storageOptions`. At this point I'm not familiar with what these options actually are, or whether they should also be incorporated into the filtering. Trying to avoid too many rabbit holes, but just something on my mind.

Reply:
Created a small cleanup [PR](https://github.com/Automattic/wp-calypso/pull/86725) to reuse the existing `PlanFeaturesForGridPlan` type.